controllers/tcproute_controller.go

```diff
@@ -0,0 +1,231 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	dataplane "github.com/kong/blixt/internal/dataplane/client"
+	"github.com/kong/blixt/pkg/vars"
+)
+
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=tcproutes,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=tcproutes/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=tcproutes/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch
+//+kubebuilder:rbac:groups=core,resources=pods/status,verbs=get
+//+kubebuilder:rbac:groups=apps,resources=daemonsets,verbs=get;list;watch
+//+kubebuilder:rbac:groups=apps,resources=daemonsets/status,verbs=get
+
+// TCPRouteReconciler reconciles a TCPRoute object
+type TCPRouteReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+
+	log logr.Logger
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *TCPRouteReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.log = log.FromContext(context.Background())
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&gatewayv1alpha2.TCPRoute{}).
+		Watches(
+			&source.Kind{Type: &appsv1.DaemonSet{}},
+			handler.EnqueueRequestsFromMapFunc(r.mapDataPlaneDaemonsetToTCPRoutes),
+		).
+		Complete(r)
+}
+
+// TCProuteReconciler reconciles TCPRoute object
+func (r *TCPRouteReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	tcproute := new(gatewayv1alpha2.TCPRoute)
+	if err := r.Get(ctx, req.NamespacedName, tcproute); err != nil {
+		if errors.IsNotFound(err) {
+			r.log.Info("object enqueued no longer exists, skipping")
+			return ctrl.Result{}, nil
+		}
+		r.log.Info("Error retrieving tcp route", "Err : ", err)
+		return ctrl.Result{}, err
+	}
+
+	isManaged, gateway, err := r.isTCPRouteManaged(ctx, *tcproute)
+	if !isManaged {
+		// TODO: enable orphan checking https://github.com/Kong/blixt/issues/47
+		return ctrl.Result{}, err
+	}
```

Comment:
```suggestion
	isManaged, gateway, err := r.isTCPRouteManaged(ctx, *tcproute)
	if err != nil {
		return ctrl.Result{}, err
	}
	if !isManaged {
		// TODO: enable orphan checking https://github.com/Kong/blixt/issues/47
		return ctrl.Result{}, nil
	}
```

Reply:
For the sake of clarity, I think we should explicitly check the error here, because, as I get from the code, isManaged can be false with err == nil (no suitable gateway has been found) or with error != nil (an error occurred when getting the gateways, and no suitable gateway has been found).

Reply:
In the code's current form, this wont make any technical difference, but I'm with you that the clarity could be better for the future. As this was copy/paste from the udproute controller, updated both in:

fa3af62270d21f1d6186ef4cac58e0b8ec13f273